third_party/nvfuser/csrc/root_domain_map.h

```diff
@@ -91,7 +91,7 @@ class TORCH_CUDA_CU_API PairwiseRootDomainMap : public RootDomainMap {
       const TensorView* producer,
       const TensorView* consumer,
       bool is_exact = false,
-      bool require_same_extent = true);
+      bool require_same_extent = false);
```

Comment:
As I mentioned before, I still think this should be true by default and selectively override the default behavior only when it's absolutely necessary.